x-pack/plugins/transform/public/app/sections/create_transform/components/step_details/step_details_form.tsx

```diff
@@ -107,6 +154,41 @@ export const StepDetailsForm: FC<Props> = React.memo(
     useEffect(() => {
       // use an IIFE to avoid returning a Promise to useEffect.
       (async function () {
+        try {
+          const { searchQuery, groupByList, aggList } = stepDefineState;
+          const pivotAggsArr = dictionaryToArray(aggList);
+          const pivotGroupByArr = dictionaryToArray(groupByList);
+          const pivotQuery = getPivotQuery(searchQuery);
+          const previewRequest = getPreviewRequestBody(
+            searchItems.indexPattern.title,
+            pivotQuery,
+            pivotGroupByArr,
+            pivotAggsArr
+          );
+
+          const transformPreview = await api.getTransformsPreview(previewRequest);
+          const properties = transformPreview.generated_dest_index.mappings.properties;
+          const datetimeColumns: string[] = [];
+
+          Object.keys(properties).forEach((col) => {
+            if (properties[col].type === 'date') {
+              datetimeColumns.push(col);
+            }
+          });
```

Comment:
```suggestion
          const datetimeColumns: string[] = Object.values(properties).filter(prop => prop.type === 'date');
```

Reply:
I needed the column names instead of the type so I modified yours slightly but it's much cleaner now. Thanks for the suggestion.
https://github.com/elastic/kibana/commit/3fe458e73e36261939a364c1a1101626b85ba7b1